### DISCUSSION TITLE
Applying impulse on click stops working

### DISCUSSION THREAD

**Discussion by anton-g** [pmndrs/use-cannon#78] [General]
I've been playing around with use-cannon, r3f and threejs and *love* how well it works together, so first off: Thanks for all the work!

Now I've run into a weird issue where `api.applyImpulse` stops working. Sometimes it works for a while, sometimes never works. The only common thing I've noticed is that it only stops working when the box(es) have settled.

I created a sandbox here: https://codesandbox.io/s/r3f-basic-demo-e4u8c

Am I just doing something weird or should this work?
Thanks!

EDIT: Did some more research and setting `allowSleep` to false "fixes" the issue, but shouldn't `applyImpulse` also "break" the sleep?

**Comment by codynova** [comment]
@anton-g Thanks for opening a discussion and doing some research, sorry for my late reply - I've been super busy lately. I will try to dig in a little more and see what Stefan's intent was, with regards to `applyImpulse` and sleeping bodies.

Paul noticed that this same issue also appears to affect Kinematic bodies.
